src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -270,10 +270,18 @@ class ReferralService(
       )
     }
 
-    update.serviceCategoryIds?.let {
-      referral.selectedServiceCategories = serviceCategoryRepository.findByIdIn(it)
+    update.serviceCategoryIds?.let { serviceCategoryIds ->
```

Comment:
happy with these changes. however i would like a comment in this block to describe what's going on here. i don't think it's immediately obvious when reading the code.

also a couple of really tiny things:

1. `referral.complexityLevelIds?.filter { complexityLevel -> serviceCategoryIds.contains(complexityLevel.key) }` could be simplified to `referral.complexityLevelIds?.filterKeys { serviceCategoryId -> serviceCategoryIds.contains(serviceCategoryId) }`

2. you can combine `save` and `flush` into `saveAndFlush(referral)`

Reply:
I've done as you've suggested